pelix/framework.py

```diff
@@ -625,13 +625,14 @@ def __init__(self, properties=None):
             # Use a copy of the properties, to avoid external changes
             self.__properties = properties.copy()
 
-        # Generate a framework instance UUID, if needed
+        # Generate and set a framework instance UUID, if needed
         framework_uid = self.__properties.get(FRAMEWORK_UID)
         if not framework_uid:
             framework_uid = str(uuid.uuid4())
-
         # Normalize the UID: it must be a string
-        self.__properties[FRAMEWORK_UID] = str(framework_uid)
+        self.__properties[FRAMEWORK_UID] = framework_uid
```

Comment:
``framework_uid`` should be converted to ``str`` before this line, to ensure that line 629 didn't return something else (like an integer or UUID object).
This would remove the need to call ``str()`` on line 631, and ensure the type on a single line